src/crispy/utils.h

```diff
@@ -223,7 +223,9 @@ std::string joinHumanReadableQuoted(std::vector<T> const& list, U sep = ", ")
 }
 
 template <typename T, typename Callback>
-constexpr inline bool split(std::basic_string_view<T> text, T delimiter, Callback const& callback)
+constexpr bool split(std::basic_string_view<T> text,
+                     T delimiter,
+                     Callback const& callback) noexcept(noexcept(callback(std::basic_string_view<T> {})))
```

Comment:
hm, why do we want to ensure and check `noexcept` specifier here?

better to restrict using `std::is_invocable` or `std::is_nothrow_invocable` if we need to make sure that the code is exceptions free

Reply:
@Yaraslaut not ensure, but check, i.e., if callback is noexcept, then split should be noexcept. That was the plan. In fact, i was juts curious, i just wanted to see how it is to actually declare a function like this. There is no deeper motivation nor actual need for this other than "I just wanted to have it done once in a lifetime" :D